compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -21,20 +21,35 @@ import transform.SymUtils._
 
 import scala.collection.mutable
 import scala.annotation.{ threadUnsafe => tu, tailrec }
+import scala.jdk.CollectionConverters._
 import scala.PartialFunction.condOpt
+import typer.ImportInfo.withRootImports
 
 import dotty.tools.dotc.{semanticdb => s}
 import dotty.tools.io.{AbstractFile, JarArchive}
+import dotty.tools.dotc.util.Property
+import dotty.tools.dotc.semanticdb.DiagnosticOps.*
+import scala.util.{Using, Failure, Success}
+
 
 /** Extract symbol references and uses to semanticdb files.
  *  See https://scalameta.org/docs/semanticdb/specification.html#symbol-1
  *  for a description of the format.
- *  TODO: Also extract type information
+ *
+ *  Here, we define two phases for "ExtractSemanticDB", "PostTyper" and "PostInlining".
+ *
+ *  The "PostTyper" phase extracts SemanticDB information such as symbol
+ *  definitions, symbol occurrences, type information, and synthetics
+ *  and write .semanticdb file.
+ *
+ *  The "PostInlining" phase extracts diagnostics from "ctx.reporter" and
+ *  attaches them to the SemanticDB information extracted in the "PostTyper" phase.
+ *  We need to run this phase after the "CheckUnused.PostInlining" phase
+ *  so that we can extract the warnings generated by "-Wunused".
  */
-class ExtractSemanticDB extends Phase:
-  import Scala3.{_, given}
+class ExtractSemanticDB private (phaseMode: ExtractSemanticDB.PhaseMode, suffix: String) extends Phase:
```

Comment:
you can use `phaseMode.toString` instead
```suggestion
class ExtractSemanticDB private (phaseMode: ExtractSemanticDB.PhaseMode) extends Phase:
```